scripts/west_commands/runners/probe_rs.py

```diff
@@ -88,7 +89,15 @@ def do_flash(self, **kwargs):
         download_args = []
         if self.erase:
             download_args += ['--chip-erase']
-        download_args += [self.elf_name]
+
+        if self.hex_file is not None:
+            download_args += ['--binary-format', 'hex']
+            download_args += [self.hex_file]
+        else:
+            download_args += ['--binary-format', 'elf']
+            download_args += [self.elf_name]
+            if self.elf_name is None:
+                raise ValueError(f'Cannot flash; elf file required.')
```

Comment:
Move this check to right after the `else` line?

```suggestion
        else:
            if self.elf_name is None:
                raise ValueError(f'Cannot flash; elf file required.')
            download_args += ['--binary-format', 'elf']
            download_args += [self.elf_name]
```